lib/cocoapods/target/build_settings.rb

```diff
@@ -489,7 +486,7 @@ def initialize(target, test_xcconfig)
         # @return [Xcodeproj::Xconfig]
         define_build_settings_method :xcconfig, :memoized => true do
           xcconfig = super()
-          xcconfig.merge(pod_target_xcconfig)
+          xcconfig.merge(merged_pod_target_xcconfigs)
```

Comment:
`merged_pod_target_xcconfigs` was unused. With this change `pod_target_xcconfig` is now unused so it's deleted. Not sure if it was copy pasta or what the intention was here and `pod_target_xcconfig_values_by_consumer_by_key` was never invoked.